sklearn/cluster/_feature_agglomeration.py

```diff
@@ -38,10 +37,7 @@ def transform(self, X):
         """
         check_is_fitted(self)
 
-        X = check_array(X)
-        if len(self.labels_) != X.shape[1]:
```

Comment:
We're assuming that the invariance of len(self.labels_) == X_train.shape[1] is enforced elsewhere, right? I guess this was only a workaround for not having ``n_features_in_`` anyway.